### PR TITLE
Add dnvm uninstall

### PR DESCRIPTION
**Pull request by BrennanConroy**
#254 
@anurse 

Still need to do powershell.

Currently uninstall is like install, you give it a version, which uses the normal defaults for choosing dnx to uninstall or you can be more specific. An idea was to allow version to be pattern matched, i.e. dnvm uninstall 1.0.0-beta6-*

### CODE REVIEW

**Comment by dougbu** [comment]
Should the `uninstall` command
1. Also remove any alias for the removed runtime?
2. Special case removing the active runtime? E.g.
  * Remove the runtime's directory from `$PATH`?
  * Error out and do nothing?
  * Output a warning but do everything?
  * Get all interactive and ask the user what to do?

**Comment by muratg** [comment]
In general it looks good to me. @anurse can you have a look?

**Comment by muratg** [comment]
:watch:

**Comment by BrennanConroy** [comment]
Shell should be good to go @anurse could you take a look?
I'll be fixing powershell half tomorrow

File: src/dnvm.sh [hunk 1]
```diff
@@ -676,6 +683,79 @@ dnvm()
...
+                if [[ $1 == "-r" || $1 == "-runtime" ]]; then
+                    local runtime=$2
+                    shift
+                elif [[ $1 == "-a" || $1 == "-arch" ]]; then
+                    local architecture=$2
+                    shift
+                elif [[ $1 == "-OS" ]]; then
+                    local os=$2
+                    shift
+		elif [[ $1 == "-y" ]]; then
```
Comment (anurse) [inline src/dnvm.sh]: Is this flag documented in the help text above?

File: test/sh/tests/uninstall/Uninstalling removes dnx.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+source $COMMON_HELPERS
+source $_DNVM_PATH
+
+$_DNVM_COMMAND_NAME install $_TEST_VERSION
+
+OUTPUT=$($_DNVM_COMMAND_NAME uninstall $_TEST_VERSION)
+echo $OUTPUT | grep "Removed $DNX_USER_HOME/runtimes/dnx-mono.$_TEST_VERSION" || die "expected message was not reported"
```
Comment (anurse) [inline test/sh/tests/uninstall/Uninstalling removes dnx.sh]: Can you test that the DNX was actually removed as well?

**Comment by anurse** [comment]
:watch: @BrennanConroy - Just some small comments

**Comment by BrennanConroy** [comment]
Verbal :shipit:

**Comment by muratg** [comment]
Yes!

**Comment by anurse** [comment]
:+1:

**Pull request merged**